core/lib/src/codegen.rs

```diff
@@ -19,6 +19,8 @@ pub struct StaticRouteInfo {
     pub handler: StaticHandler,
     /// The route's rank, if any.
     pub rank: Option<isize>,
+    /// The states requested by this route
+    pub states: [Option<std::any::TypeId>; 16],
```

Comment:
I think `&'static [Option<std::any::TypeId>]` should work here, simplifying some other code as well.